src/pyFAI/geometry/core.py

```diff
@@ -2767,99 +2767,131 @@ def get_shape(self, shape=None):
+    get_poni2 = deprecated(poni2.fget, reason="use property", since_version="2025.09")
+    set_poni2 = deprecated(poni2.fset, reason="use property", since_version="2025.09")
+
 
-    poni2 = property(get_poni2, set_poni2)
+    @property
+    def rot1(self):
+        return self._rot1
 
-    def set_rot1(self, value):
-        if isinstance(value, float):
+    @rot1.setter
+    def rot1(self, value):
+        if isinstance(value, float):  # TODO: Is this still necessary?
             self._rot1 = value
         elif isinstance(value, (tuple, list)):
             self._rot1 = float(value[0])
         else:
             self._rot1 = float(value)
         self.reset()
 
-    def get_rot1(self):
-        return self._rot1
+    # deprecated compatibility layer
+    get_rot1 = deprecated(rot1.fget, reason="use property", since_version="2025.09")
+    set_rot1 = deprecated(rot1.fset, reason="use property", since_version="2025.09")
+
 
-    rot1 = property(get_rot1, set_rot1)
+    @property
+    def rot2(self):
+        return self._rot2
 
-    def set_rot2(self, value):
-        if isinstance(value, float):
+    @rot2.setter
+    def rot2(self, value):
+        if isinstance(value, float):  # TODO: Is this still necessary?
             self._rot2 = value
         elif isinstance(value, (tuple, list)):
             self._rot2 = float(value[0])
         else:
             self._rot2 = float(value)
         self.reset()
 
-    def get_rot2(self):
-        return self._rot2
+    # deprecated compatibility layer
+    get_rot2 = deprecated(rot2.fget, reason="use property", since_version="2025.09")
+    set_rot2 = deprecated(rot2.fset, reason="use property", since_version="2025.09")
 
-    rot2 = property(get_rot2, set_rot2)
+    @property
+    def rot3(self):
+        return self._rot3
 
-    def set_rot3(self, value):
-        if isinstance(value, float):
+    @rot3.setter
+    def rot3(self, value):
+        if isinstance(value, float):  # TODO: Is this still necessary?
             self._rot3 = value
         elif isinstance(value, (tuple, list)):
             self._rot3 = float(value[0])
         else:
             self._rot3 = float(value)
         self.reset()
 
-    def get_rot3(self):
-        return self._rot3
+    # deprecated compatibility layer
+    get_rot3 = deprecated(rot3.fget, reason="use property", since_version="2025.09")
+    set_rot3 = deprecated(rot3.fset, reason="use property", since_version="2025.09")
 
-    rot3 = property(get_rot3, set_rot3)
 
-    def set_wavelength(self, value):
+    @property
+    def wavelength(self):
+        return self._wavelength
+
+    @wavelength.setter
+    def wavelength(self, value):
         "Set the wavelength in meter!"
         old_wl = self._wavelength
-        if isinstance(value, float):
+        if isinstance(value, float):  # TODO: Is this still necessary?
             self._wavelength = value
         elif isinstance(value, (tuple, list)):
             self._wavelength = float(value[0])
```

Comment:
According to the documentation, `wavelength` should be of type `float`. 
https://pyfai.readthedocs.io/en/stable/api/geometry/index.html#pyFAI.geometry.core.Geometry.__init__
Is it necessary to keep the check for `tuple`, `list`?

Reply:
I would deprecate the usage of lists/tuple